src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -842,6 +948,35 @@ private static bool ShouldIncludePropertyForFastPath(PropertyGenerationSpec prop
                 return true;
             }
 
+            private static string GetParameterizedCtorInvocationFunc(TypeGenerationSpec typeGenerationSpec)
+            {
+                ParameterGenerationSpec[] parameters = typeGenerationSpec.CtorParamGenSpecArray;
+                int paramCount = parameters.Length;
+                Debug.Assert(paramCount != 0);
+
+                if (paramCount > JsonConstants.MaxParameterCount)
+                {
+                    return "null";
```

Comment:
It seems like an error of some sort should be provided, at least if that's what we do for reflection cases. Can we use ReportDiagnostic?

Reply:
Yup we can use `ReportDiagnostic` for this since constructor selection can only be influenced with design-time config. Right now we throw a runtime exception instead, just like the dynamic serializer.

Reply:
A run-time exception sounds better since it is consistent with the dynamic serializer.